Lib/multiprocessing/shared_memory.py

```diff
@@ -0,0 +1,677 @@
+from functools import reduce
+import mmap
+from .managers import DictProxy, SyncManager, Server
+import os
+import random
+import struct
+import sys
+try:
+    from _posixshmem import _PosixSharedMemory, Error, ExistentialError, O_CREX
+except ImportError as ie:
+    if os.name != "nt":
+        # On Windows, posixshmem is not required to be available.
+        raise ie
+    else:
+        _PosixSharedMemory = object
+        class ExistentialError(BaseException): pass
+        class Error(BaseException): pass
+        O_CREX = -1
+
+
+class WindowsNamedSharedMemory:
+
+    def __init__(self, name, flags=None, mode=None, size=None, read_only=False):
+        if name is None:
+            name = f'wnsm_{os.getpid()}_{random.randrange(100000)}'
```

Comment:
`random.randrange(100000)`
Why use a random number here?

Reply:
Replaced by Neil's patch with a stronger solution in GH-11816.